.github/workflows/ci_testing.yml

```diff
@@ -57,6 +57,24 @@ jobs:
           fail_ci_if_error: true
           verbose: true
         if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.9'
+      - name: Build Docs
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
+        run: |
+          conda install doxygen
+          cd docs
+          doxygen ./Doxyfile
+          make html SOURCEDIR=.
+      - name: Set AWS credentials for upload
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
+        uses: aws-actions/configure-aws-credentials@0e613a0980cbf65ed5b322eb7a1e075d28913a83
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - name: Upload to S3
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
+        run: |
+          aws s3 sync --delete docs/build/html s3://asc-docs/ale
```

Comment:
I am worried that this builds and push docs for every PR that comes through without being merged. Ideally we would only want to deploy docs once merged into main

Reply:
Hm ok so I need to set some sort of condition that only builds/pushes docs if it's a PR being merged into `main`.